sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -617,21 +617,30 @@ public void onCallFailed(Exception e) {
 
     private synchronized AppCenterFuture<Document<Void>> instanceDelete(final String partition, final String documentId) {
         final DefaultAppCenterFuture<Document<Void>> result = new DefaultAppCenterFuture<>();
-        getTokenAndCallCosmosDbApi(
-                partition,
-                result,
-                new TokenExchange.TokenExchangeServiceCallback() {
+        if (mNetworkStateHelper.isNetworkConnected()) {
+            getTokenAndCallCosmosDbApi(
+                    partition,
+                    result,
+                    new TokenExchange.TokenExchangeServiceCallback() {
 
-                    @Override
-                    public void callCosmosDb(TokenResult tokenResult) {
-                        callCosmosDbDeleteApi(tokenResult, documentId, result);
-                    }
+                        @Override
+                        public void callCosmosDb(TokenResult tokenResult) {
+                            callCosmosDbDeleteApi(tokenResult, documentId, result);
+                        }
 
-                    @Override
-                    public void completeFuture(Exception e) {
-                        Storage.this.completeFuture(e, result);
-                    }
-                });
+                        @Override
+                        public void completeFuture(Exception e) {
+                            Storage.this.completeFuture(e, result);
+                        }
+                    });
+        } else {
+            boolean isWriteSucceed = mLocalDocumentStorage.writeDelete(partition, documentId);
```

Comment:
This causes strict mode issues to access disk on the calling thread which is most of the time the UI thread.

Please catch up with develop and look at how we use async here in similar methods.